agent/handlers/v2/response.go

```diff
@@ -128,9 +134,35 @@ func NewTaskResponse(taskARN string,
 		resp.Containers = append(resp.Containers, containerResponse)
 	}
 
+	if propagateTags {
+		propagateTagsToMetadata(state, ecsClient, containerInstanceArn, taskARN, resp)
+	}
+
 	return resp, nil
 }
 
+func propagateTagsToMetadata(state dockerstate.TaskEngineState, ecsClient api.ECSClient, containerInstanceArn, taskARN string, resp *TaskResponse) {
+	containerInstanceTags, err := ecsClient.GetResourceTags(containerInstanceArn)
+	if err == nil {
+		resp.ContainerInstanceTags = make(map[string]string)
+		for _, tag := range containerInstanceTags {
+			resp.ContainerInstanceTags[*tag.Key] = *tag.Value
+		}
+	} else {
+		seelog.Errorf("Could not get container instance tags for %s: %s", containerInstanceArn, err.Error())
```

Comment:
If we are unable to retrieve tags, we'd better return an error instead of failing silently.

You can check the tags in the tags handler and return error if it doesn't exist.

Reply:
There are 2 calls to ListTagsForResource and either could fail based on customer setting for long-arn. I think the correct behavior is to fail silently